### ISSUE TITLE
HACS Compatibility

### ISSUE THREAD

**Issue by dshanske** [calmor15014/HA-Component-mythtv-frontend#39]
Would make it easier to install/update.

https://hacs.xyz/docs/publish/start

**Comment by calmor15014** [comment]
Thanks.  The main reason we didn't push this across the finish line to get included into a Home Assistant official release is that there was not a pip install path for the MythTV python utilities that we were working on with their creator; pip requires more maintenance than anyone wanted to do, me included (you see how often I get in here as it is...).  As a result, the installation required a manual import that would occasionally change, at least at the time.  

As of MythTV v30, the Python library is included with MythTV. However, most people aren't running Home Assistant on the same machine as their MythTV install, so that's not a likely path for easy distribution.

Option two was to basically replicate the parts of the python bindings that we needed within our own code. In some ways, that goes against what Home Assistant is supposed to be about - importing libraries when available and not reinventing the wheel.  But, the calls are relatively simple http calls in the grand scheme of things, and well defined on the MythTV site, and the library is available but not easily automated for installation... so it seems reasonable to replicate them.  That work needs done first though before we can propose to add it to Home Assistant as an easy install.

**Comment by dshanske** [comment]
I'm a PHP person, beginning to think I should pick up enough Python for HA

**Comment by calmor15014** [comment]
It all looks the same to me after a while. I have to Google everything when I switch projects anyway. Too many hobbies, not enough time. I know it's not php when there are no $ all over the place.

The services API is [here](https://github.com/billmeek/MythTVServicesAPI) and Bill has done a great job with it. I think we're just using `send.py` from it, and it's much more solid code than what I've ever managed to cobble together... which is why I didn't want to recreate it.  We probably don't _need_ all of those checks and balances and can just hit the API addresses directly (I've done it with curl in bash scripts), but it''s nice to have them.

**Comment by RXM307** [comment]
> Thanks. The main reason we didn't push this across the finish line to get included into a Home Assistant official release is that there was not a pip install path for the MythTV python utilities that we were working on with their creator; pip requires more maintenance than anyone wanted to do, me included (you see how often I get in here as it is...). As a result, the installation required a manual import that would occasionally change, at least at the time.
> 
> As of MythTV v30, the Python library is included with MythTV. However, most people aren't running Home Assistant on the same machine as their MythTV install, so that's not a likely path for easy distribution.
> 
> Option two was to basically replicate the parts of the python bindings that we needed within our own code. In some ways, that goes against what Home Assistant is supposed to 'be about - importing libraries when available and not reinventing the wheel. It also means that we're fully on the hook for making sure they keep working for future MythTV releases and not just maintenance for deprecating changes in Home Assistant releases. But, the calls are relatively simple http calls in the grand scheme of things, and well defined on the MythTV site, and the library is available but not easily automated for installation... so it seems reasonable to replicate them. That work needs done first though before we can propose to add it to Home Assistant as an easy install.

I'm running 0.30 and getting around this by exporting the mythtv libraries into my home assistant docker container but like you said not everyone is running these on the same server as mythtv, however many integrations in HACS require some work outside of HA to get them working like SOAP application servers etc so i dont see any issue as long as the requirement is documented (anyone running this will have access to the required libraries)

ie. from my docker-compose.yml
        - /etc/homeassistant/mythtv_services_api:/usr/local/lib/python3.7/site-packages/mythtv_services_api
        - /etc/homeassistant/mythtv_services_api-0.1.9.dist-info:/usr/local/lib/python3.7/site-packages/mythtv_services_api-0.1.9.d

**Comment by calmor15014** [comment]
> ie. from my docker-compose.yml
> \- /etc/homeassistant/mythtv_services_api:/usr/local/lib/python3.7/site-packages/mythtv_services_api
> \- /etc/homeassistant/mythtv_services_api-0.1.9.dist-info:/usr/local/lib/python3.7/site-packages/mythtv_services_api-0.1.9.d

Interesting way to do it.  For me, I'm running both HA and MythTV in separate LXC containers on the same Proxmox hypervisor, so I'm sure there's a way I could patch it through.  But, a lot of people running on RPis or in Nabu Casa aren't going to have easy access to those files.

Realistically, the best way to go about it would be to recreate the wheel for the frontend API services, which isn't _hard_, it just needs done and tested.  Could install something on the MythTV server that Home Assistant could more easily talk to, since the Python libraries are in fact on the MythTV server already... but that seems like more work for minimal if any benefit, when a SOAP API interface is already provided.

Because of deprecation changes in the latest HA, I'm updating my HA dev container fully and will try to fix that and, while I'm in there, see what would be needed to integrate all of this into one file package for HACS.

**Comment by lindsaymarkward** [comment]
I've been updating the services API to go on PyPI.
From there, I am thinking of submitting this platform to HA directly rather than go through HACS.

**Comment by calmor15014** [comment]
Agreed.  Let's close this for now and get it into HA direct.
